File: lib/MessageBroker/dispatchers/RabbitMQ/routings/PatternMatching/PatternMatchingRabbitMQRoom.rb
require_relative '../BaseRabbitMQRoom'

class PatternMatchingRabbitMQRoom < BaseRabbitMQRoom
  def initialize name:, channel:, exchange:
    queue = channel.queue(name, :auto_delete => true).bind(exchange, :routing_key => name)

    super(queue: queue)
  end
end

File: js/lib/MessageBroker/dispatchers/RabbitMQ/routings/PatternMatching/PatternMatchingRabbitMQRoom.js
const BaseRabbitMQRoom = require("../BaseRabbitMQRoom");

class PatternMatchingRabbitMQRoom extends BaseRabbitMQRoom {
  constructor({ channel, queue }) {
    super({ channel, queue });
  }

  static async create({ name, topicName, channel }) {
    try {
      let q = await channel.assertQueue(); 
      channel.bindQueue(q.queue, topicName, name);

      return new PatternMatchingRabbitMQRoom({ channel, queue: q.queue })
    } catch (error) {
      throw new Error("Pattern matching RabbitMQ room creation error");
    }
  }
}

module.exports = PatternMatchingRabbitMQRoom